app/src/main/java/com/checkin/app/checkin/home/HomeRepository.kt

```diff
@@ -0,0 +1,42 @@
+package com.checkin.app.checkin.home
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.LiveData
+import com.checkin.app.checkin.data.BaseRepository
+import com.checkin.app.checkin.data.network.ApiClient
+import com.checkin.app.checkin.data.network.ApiResponse
+import com.checkin.app.checkin.data.network.RetrofitLiveData
+import com.checkin.app.checkin.data.network.WebApiService
+import com.checkin.app.checkin.data.resource.NetworkBoundResource
+import com.checkin.app.checkin.data.resource.Resource
+import com.checkin.app.checkin.home.model.ClosedSessionDetailsModel
+import com.checkin.app.checkin.home.model.CustomerClosedSessionModel
+import com.checkin.app.checkin.utility.SingletonHolder
+
+class HomeRepository private constructor(context: Context) : BaseRepository() {
+    private val mWebService: WebApiService = ApiClient.getApiService(context)
+
+    val customerClosedSessionList: LiveData<Resource<List<CustomerClosedSessionModel>>> =
```

Comment:
You've written the code wrong. Repository is never supposed to have data attributes, since it's not supposed to store anything. The only attribute ever defined is the web service.

Reply:
Now you've created a **immutable constant field** in repository.
This is different from regular methods or using delegated get() for val fields... Now it's storing the resulting livedata in memory. So your network call only happens once and then it's unintentionally cached in memory.